core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -45,27 +46,27 @@ private[log] case object LogCleaningPaused extends LogCleaningState
  *  While a partition is in the LogCleaningPaused state, it won't be scheduled for cleaning again, until cleaning is
  *  requested to be resumed.
  */
-private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[TopicPartition, Log]) extends Logging with KafkaMetricsGroup {
+private[log] class LogCleanerManager(val logDirs: ArrayBuffer[File], val logs: Pool[TopicPartition, Log]) extends Logging with KafkaMetricsGroup {
```

Comment:
We need to pass `LogManager.liveLogDirs` to `LogCleanerManager` by reference and be able to propagate the change in `LogManager.liveLogDirs` to `LogCleanerManager`. Therefore we need to change its type to ArrayBuffer so that it is mutable. Does this make sense?